OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1

```diff
@@ -237,7 +262,21 @@ function Import-ServerConfig {
 
     if (Test-OctopusVersionSupportsShowConfiguration) {
         $rawConfig = & $octopusServerExePath show-configuration --format=json-hierarchical --noconsolelogging --console --instance $InstanceName
-        $config = $rawConfig | ConvertFrom-Json
+
+        # handle a specific error where an exception in registry migration finds its way into the json-hierarchical output
+        # Refer to Issue #179 (https://github.com/OctopusDeploy/OctopusDSC/issues/179)
+
+        if(Test-ValidJson $rawConfig) {
+            $config = $rawConfig | ConvertFrom-Json
+        } else {
+            Write-Warning "Invalid json encountered in show-configuration; attempting to clean up."
+            $cleanedUpConfig = Get-CleanedJson $rawConfig
+            if(Test-ValidJson $cleanedUpConfig ) {
+                $config = $cleanedUpConfig | ConvertFrom-Json
+            } else {
+                Write-Error "Attempted to cleanup bad JSON and failed. String we attempted to parse was`r`n`r`n$rawConfig"
```

Comment:
Could this include a connection string? Or a master key?
Are we masking?

Reply:
Removed the bit of potentially leaky code